homeassistant/components/sia/sia_entity_base.py

```diff
@@ -104,33 +105,42 @@ def async_handle_event(self, sia_event: SIAEvent) -> None:
         if int(sia_event.ri) not in (self.zone, SIA_HUB_ZONE):
             return
         self._attr_extra_state_attributes.update(get_attr_from_sia_event(sia_event))
-        state_changed = self.update_state(sia_event)
-        if state_changed:
-            self.async_reset_availability_cb()
+        update_ping_interval_cb = self.update_state(sia_event)
+        if update_ping_interval_cb:
+            self._attr_available = True
+            self._cancel_post_interval_update_cb()
+            self.async_create_post_interval_update_cb()
         self.async_write_ha_state()
 
     @abstractmethod
     def update_state(self, sia_event: SIAEvent) -> bool:
-        """Do the entity specific state updates."""
+        """Do the entity specific state updates.
 
-    @callback
-    def async_reset_availability_cb(self) -> None:
-        """Reset availability cb by cancelling the current and creating a new one."""
-        self._attr_available = True
-        if self._cancel_availability_cb:
-            self._cancel_availability_cb()
-        self.async_create_availability_cb()
-
-    def async_create_availability_cb(self) -> None:
-        """Create a availability cb and return the callback."""
-        self._cancel_availability_cb = async_call_later(
+        Return True if the interval callback needs to be updated.
+        """
+
+    def async_create_post_interval_update_cb(self) -> None:
+        """Create a port interval update cb and return the callback."""
+        self._post_interval_update_cb_canceller = async_call_later(
             self.hass,
             get_unavailability_interval(self.ping_interval),
-            self.async_set_unavailable,
+            self.async_post_interval_update,
         )
 
     @callback
-    def async_set_unavailable(self, _) -> None:
-        """Set unavailable."""
+    def async_post_interval_update(self, _) -> None:
+        """Set unavailable or update state after a ping interval.
+
+        Since not all types of entities that inherit from SIABaseEntity have _attr_is_on, this function uses the setattr instead of a direct assignment.
+        """
+        if self.entity_description.set_state_not_availability:
+            setattr(self, "_attr_is_on", False)
+            self.async_write_ha_state()
+            return
```

Comment:
This looks quite hacky. Would it be cleaner to have a separate class for connectivity sensor (extending `SIABinarySensor`) that would override this method?